### DISCUSSION TITLE
decryptBlob security

### DISCUSSION THREAD

**Discussion by henke37** [airsdk/Adobe-Runtime-Support#2153] [General]
The new decryptBlob method is interesting, but I wonder how secure it is.

Let's not beat around the bush, the function does not take a key, so it's more of an encoding. I'd assume that the key is derived somehow? How secure is this derivation? What are its inputs?

**Comment by ajwfrost** [comment]
Yes exactly, we derive the key and if someone spent a while with a decompiler and wading through the assembly code, they could work out how to generate this.

I'd be open to ideas for how else to secure this, the challenge is always going to be how a key can be separately delivered. There's no benefit I think to having it in the AS3 source code..

Maybe we should make it an option to also provide a key - but that key would need to be provided somehow (a) to the compiler (which can be through a command-line argument so it's not part of the source code/metadata) and (b) to the user so that they can enter this.. perhaps having logged in to a server component that then provides the key via https?

Would be interested in your thoughts :-)

thanks
